sdks/java/harness/src/main/java/org/apache/beam/fn/harness/state/StateFetchingIterators.java

```diff
@@ -84,10 +83,7 @@ private StateFetchingIterators() {}
       StateRequest stateRequestForFirstChunk,
       Coder<T> valueCoder) {
     return new CachingStateIterable<>(
-        (Cache<StateKey, Blocks<T>>) cache,
-        beamFnStateClient,
-        stateRequestForFirstChunk,
-        valueCoder);
+        (Cache<Object, Blocks<T>>) cache, beamFnStateClient, stateRequestForFirstChunk, valueCoder);
```

Comment:
I would suggest using a static object that is weighted that is internal. Using `null` hides useful information when debugging the caches contents. Please also update the method comment saying that the state cache should be namespaced for this object to prevent collisions (like BagUserState).

Like 
```
private static class IterableCacheKey implements Weighted {
  static INSTANCE = new IterableCacheKey();
  getWeight() {
    return 0;
  }
}
```

Reply:
Done.
Instead of weighting the singleton instance I report zero since the weight will be counted many times in separate subcaches. Added comment with reasoning.